tests/unit/h/schemas/forms/admin/group_test.py

```diff
@@ -17,6 +18,18 @@ class TestAdminGroupSchema:
     def test_it_allows_with_valid_data(self, group_data, bound_schema):
         bound_schema.deserialize(group_data)
 
+    def test_it_raises_if_csrf_token_missing(self, group_data, bound_schema):
+        del bound_schema.bindings["request"].headers["X-CSRF-Token"]
+
+        with pytest.raises(BadCSRFToken):
+            bound_schema.deserialize(group_data)
+
+    def test_it_raises_if_csrf_token_wrong(self, group_data, bound_schema):
+        bound_schema.bindings["request"].headers["X-CSRF-Token"] = "foobar"
+
+        with pytest.raises(BadCSRFToken):
+            bound_schema.deserialize(group_data)
```

Comment:
By default Pyramid will read the CSRF token from either a `csrf_token` form field or an `X-CSRF-Token` header. In practice our pages use the form field. But our tests use the header (via the `pyramid_csrf_request` fixture).